server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -79,8 +79,6 @@
         this.asList = Collections.singletonList(this);
         assert expectedShardSize == UNAVAILABLE_EXPECTED_SHARD_SIZE || state == ShardRoutingState.INITIALIZING ||
             state == ShardRoutingState.RELOCATING : expectedShardSize + " state: " + state;
-        assert expectedShardSize >= 0 || state != ShardRoutingState.INITIALIZING || state != ShardRoutingState.RELOCATING :
```

Comment:
why remove this?

Reply:
This seemed (mathematically it is) already covered by the above `assert`, but I see you point about documentation above. I'll add it back :)